tests/core/data_storage/test_in_memory_storage.py

```diff
@@ -0,0 +1,113 @@
+import functools
+
+import numpy
+
+from openfisca_core import data_storage
+from openfisca_core import periods
+
+import pytest
+
+
+@pytest.fixture
+def storage():
+    return data_storage.InMemoryStorage
+
+
+@pytest.fixture
+def eternal_storage(storage):
+    return functools.partial(storage, is_eternal = True)
+
+
+@pytest.fixture
+def period():
+    return periods.period("2020")
+
+
+@pytest.fixture
+def value():
+    return numpy.array([1])
+
+
+def test___init__(storage):
+    result = storage()
+
+    assert result
+
+
+def test___init__when_is_eternal(eternal_storage):
+    result = eternal_storage()
+
+    assert result.is_eternal
+
+
+def test_put(storage, period, value):
+    storage = storage()
+    storage.put(value, period)
+
+    result = storage.get(period)
+
+    assert result == value
+
+
+def test_put_when_is_eternal(eternal_storage, value):
+    storage = eternal_storage()
+    storage.put(value, "foo")
+
+    result = storage.get("bar")
+
+    assert result == value
+
+
+def test_delete(storage, period, value):
+    storage = storage()
+    storage.put(value, period)
+    storage.put(value, period.last_year)
+    storage.delete(period)
+
+    result = storage.get(period), storage.get(period.last_year)
+
+    assert result == (None, value)
+
+
+def test_delete_when_period_is_not_specified(storage, period, value):
+    storage = storage()
+    storage.put(value, period)
+    storage.put(value, period.last_year)
+    storage.delete()
+
+    result = storage.get(period)
+
+    assert not result
+
+
+def test_delete_when_is_eternal(eternal_storage, value):
+    storage = eternal_storage()
+    storage.put(value, "qwerty")
+    storage.put(value, "azerty")
+    storage.delete("asdf1234")
```

Comment:
See above; better to use periods.

Reply:
I'm trying to expose the uselessness of some parts the current implementation to other contributors who can come after me. Would it be clearer if I add an explanation ? The objective of the test is to say "whatever you do, as long as it is an eternal definition period we're dealing with, won't have no effect whatsoever, so wonder why does this code is necessary at all".

Reply:
OK, I get where you're coming from.

Keep in mind that all the information I have when you ping me for review is "I'm being asked for a review". If there is a type of feedback that's more (or less) useful to you, I'm happy to consider that before I start reviewing - I always make a stop at the PR's description before diving into the code itself, so that's a useful place (for me) to have information about intent and context.

By default, I'm going to assume that the last contract we discussed still applies: i.e. "would you be happy maintaining this code as it will be after this PR".

Reply:
Thanks a lot! Yeah I think I'll have to go the extra mile for this to be mergeable then.

Reply:
I did use periods in the end.